### PR TITLE
Add Viscous Timestepping

### PR DESCRIPTION
**Pull request by jlevine18**
This update adds `get_viscous_timestep` and `get_viscous_cfl` functions to support viscosity adjustments to constant CFL mode. These functions use Hesthaven's expression for CFL calculation: 
image.

### CODE REVIEW

File: mirgecom/viscous.py [hunk 1]
```diff
@@ -221,23 +222,51 @@ def viscous_facial_flux(discr, eos, cv_tpair, grad_cv_tpair,
...
+
+    Similar to get_inviscid_timestep but there's a viscosity term.
+    """
+    from grudge.dt_utils import characteristic_lengthscales
+    from mirgecom.fluid import compute_wavespeed
+    return (
+        characteristic_lengthscales(cv.array_context, discr)
+        / (compute_wavespeed(eos, cv)
+        + get_inv_length_weighted_viscosity(discr, eos, cv))
+    )
```
Comment (MTCam) [inline mirgecom/viscous.py:249]: Let's refactor this so that we call characteristic_lengthscales only once.

File: mirgecom/viscous.py [hunk 2]
```diff
@@ -221,23 +222,51 @@ def viscous_facial_flux(discr, eos, cv_tpair, grad_cv_tpair,
...
-    return cfl * dt
-#    dt_ngf = dt_non_geometric_factor(discr.mesh)
-#    dt_gf  = dt_geometric_factor(discr.mesh)
-#    wavespeeds = compute_wavespeed(w,eos=eos)
-#    max_v = clmath.max(wavespeeds)
-#    return c*dt_ngf*dt_gf/max_v
+    from grudge.dt_utils import characteristic_lengthscales
+
+    transport = eos.transport_model()
+    mu = transport.viscosity(eos, cv)
```
Comment (MTCam) [inline mirgecom/viscous.py:233]:
```suggest
    mu = 0
    transport = eos.transport_model()
    if transport:
       mu = transport.viscosity(eos, cv)
```

File: mirgecom/viscous.py (hunk 1 above)
Comment (jlevine18) [reply mirgecom/viscous.py]: Refactored by merging `get_viscous_timestep` and `get_inv_length_weighted_viscosity` into one function.

File: mirgecom/viscous.py (hunk 2 above)
Comment (jlevine18) [reply mirgecom/viscous.py]: Expanded as above in merged function.

File: mirgecom/viscous.py [hunk 3]
```diff
@@ -221,23 +222,46 @@ def viscous_facial_flux(discr, eos, cv_tpair, grad_cv_tpair,
...
+    mu = 0
+    transport = eos.transport_model()
+    if transport:
+        mu = transport.viscosity(eos, cv)
+
+    weighted_viscosity = (mu / length_scales)
+
+    return (
+        length_scales / (compute_wavespeed(eos, cv) + weighted_viscosity)
+    )
```
Comment (MTCam) [inline mirgecom/viscous.py:244]: These little temporaries make it easier to read and understand, but we avoid them in this code whenever possible.
```suggestion
    return(
        length_scales / (compute_wavespeed(eos, cv) + mu / length_scales)
    )
```

**Review by MTCam — COMMENTED** [review]
This looks really great @jlevine18!  Thanks for your work on this.  There are just a few things to wrap up for a clean merge with the existing y1 tower, and this is good-to-go.

File: mirgecom/viscous.py [hunk 4]
```diff
@@ -221,23 +222,44 @@ def viscous_facial_flux(discr, eos, cv_tpair, grad_cv_tpair,
...
 
 
-def get_viscous_timestep(discr, eos, transport_model, cfl, cv):
-    """Routine (will) return the (local) maximum stable viscous timestep.
+def get_viscous_timestep(discr, eos, cv):
+    """Routine returns the the node-local maximum stable viscous timestep.
 
-    Currently, it's a hack waiting for the geometric_factor helpers port
-    from grudge.
+    Similar to get_inviscid_timestep but there's a viscosity term.
```
Comment (MTCam) [inline mirgecom/viscous.py:228]:
```suggestion
    Parameters
    ----------
    discr: grudge.eager.EagerDGDiscretization
        the discretization to use
    eos: mirgecom.eos.GasEOS
        An equation of state implementing the speed_of_sound method
    cv: :class:`~mirgecom.fluid.ConservedVars`
        Fluid solution

    Returns
    -------
    class:`~meshmode.dof_array.DOFArray`
        The maximum stable timestep at each node.
```

Comment (MTCam) [reply mirgecom/viscous.py:228]: Sorry, one more :)

**Pull request merged**